### PR TITLE
domain: support extract plan task handle

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: ref #41330 

Problem Summary:

### What is changed and how it works?

1. revert #41478
2. support extract handler to extract plan information from stmt_summary_history

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -132,6 +132,7 @@ type Domain struct {
 	indexUsageSyncLease   time.Duration
 	dumpFileGcChecker     *dumpFileGcChecker
 	planReplayerHandle    *planReplayerHandle
+	extractTaskHandle     *extractHandle
```
Comment (qw4990) [inline domain/domain.go:135]: Can we just merge this handler with `planReplayerHandle` to let it simpler? 
Then we can implement all features related to `plan information dumping` together.

File: domain/extract.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+		p.sqls = append(p.sqls, record.sql)
+		for _, tbl := range record.tables {
+			p.tables[tbl] = struct{}{}
+		}
+	}
+	return p
+}
+
+func (w *extractWorker) dumpExtractPlanPackage(p *extractPlanPackage) (name string, err error) {
```
Comment (qw4990) [inline domain/extract.go:221]: Better to add some comments for this function to explain how we organize the package, just like `DumpPlanReplayerInfo`.

File: domain/extract.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+	switch task.extractType {
+	case ExtractPlanType:
+		return w.extractPlanTask(ctx, task)
+	}
+	return "", nil
+}
+
+func (w *extractWorker) extractPlanTask(ctx context.Context, task *ExtractTask) (string, error) {
+	if !config.GetGlobalConfig().Instance.StmtSummaryEnablePersistent {
+		return "", errors.New("tidb_stmt_summary_enable_persistent should be enabled for extract task")
```
Comment (qw4990) [inline domain/extract.go:131]: It seems like we cannot extract any information if tidb_stmt_summary_enable_persistent is disabled. Can we dump information from the memory `stmt_summary` table and generate a warning to notify the user the dumped information may not be completed in this case?

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**